web/modules/custom/farm_page/src/Controller/FarmPageController.php

```diff
@@ -151,4 +152,46 @@ public function listFarm() {
     ];
   }
 
+  /**
+   * Build list farm.
+   *
+   * @return mixed
+   *   Return Farmr and Farm information
+   */
+  public function listProductOfFarm($uid, $categories_id = NULL) {
+    $store_id = $this->farmService->getStoreIdByUserId($uid);
+    $store = Store::load($store_id);
+    $products = $this->farmService->getProductByStore($uid, $categories_id);
+    $categories = $this->farmService->getTermOfTaxonomy('categories');
+    $address = $store->get('address')->getValue()[0];
+    $farm = [
+      'name' => $store->getName(),
+      'description' => $store->get('field_description')->getValue()[0]['value'],
+      'farm_images' => reset($this->farmService->getImagesUrlFromArray($store_images_field)),
+      'phone' => $store->get('field_store_phone')->getValue()[0]['value'],
```

Comment:
Please check for null before using getValue() (for example !empty($store->get('field_description'))) to avoid a 500 error.

Reply:
updated